### PR TITLE
 [MIG] website_form_recaptcha: Migrate to 11.0

### PR DESCRIPTION
**Pull request by bouvyd**

### CODE REVIEW

**Comment by bouvyd** [comment]
I have no clue what to do about the codecov tests...

**Comment by lasley** [comment]
> I have no clue what to do about the codecov tests...

Safe to ignore, it just means there isn't an existing coverage report in the target branch. Perfectly normal when there's no modules in the branch

File: website_form_recaptcha/data/ir_config_parameter_data.xml
Comment (yajo) [inline website_form_recaptcha/data/ir_config_parameter_data.xml]: Remove data tag and put noupdate directly in odoo tag please

File: website_form_recaptcha/models/website_form_recaptcha.py
Comment (yajo) [inline website_form_recaptcha/models/website_form_recaptcha.py]: You cannot translate strings outside a method because there's no `self.env.lang` variable to get the desired language, so if you're translating this (which is good IMHO), then do not use a class variable and use a method instead please 😊

File: website_form_recaptcha/data/ir_config_parameter_data.xml
Comment (yajo) [inline website_form_recaptcha/data/ir_config_parameter_data.xml]: [You shouldn't remove the groups line](https://github.com/OCA/website/pull/395/commits/ec1240455f9b4ba1c51b922eff0a3cc96a9621ed#diff-567885b3b5da5f9ff0f0bdf646534a4fL20), this secret must not be world-readable.

**Review by yajo — CHANGES_REQUESTED** [review]
Please also squash all transbot commits together, including dbcf6f7

File: website_form_recaptcha/data/ir_config_parameter_data.xml
Comment (bouvyd) [reply website_form_recaptcha/data/ir_config_parameter_data.xml]: The `group_ids` field is gone as of odoo/odoo@d0ca2d115e8b40308e1ff83bf9e7cc874762fbff, every `ir.config_parameter` must be accessed with `sudo()` now.

File: website_form_recaptcha/data/ir_config_parameter_data.xml
Comment (bouvyd) [reply website_form_recaptcha/data/ir_config_parameter_data.xml]: :+1:

File: website_form_recaptcha/models/website_form_recaptcha.py
Comment (bouvyd) [reply website_form_recaptcha/models/website_form_recaptcha.py]: Good catch

File: website_form_recaptcha/data/ir_config_parameter_data.xml
Comment (pedrobaeza) [reply website_form_recaptcha/data/ir_config_parameter_data.xml]: Thanks for the info. Added to the migration wiki.

**Comment by pedrobaeza** [comment]
About merging "OCA Transbot ..." commits, we have this reference guide: https://github.com/OCA/maintainer-tools/wiki/Merge-commits-in-pull-requests

File: website_form_recaptcha/models/website_form_recaptcha.py
Comment (lasley) [reply website_form_recaptcha/models/website_form_recaptcha.py]: TIL

**Review by lasley — APPROVED** [review]
Thanks @dbo-odoo - LGTM outside of @Yajo comments

**Comment by bouvyd** [comment]
Review changes suggested by @Yajo done!

File: website_form_recaptcha/data/ir_config_parameter_data.xml
Comment (yajo) [inline website_form_recaptcha/data/ir_config_parameter_data.xml]: Please use coherent indentation with 4 spaces 😄

File: website_form_recaptcha/models/website_form_recaptcha.py
Comment (yajo) [inline website_form_recaptcha/models/website_form_recaptcha.py]: It seems more sensible IMHO to have a `_get_error_message(self, errorcode)` method instead. Not a requirement.

**Review by yajo — CHANGES_REQUESTED** [review]
A couple of nitpicks more to go, and please [check the linter](https://travis-ci.org/OCA/website/jobs/317368212#L582).

Thanks!

File: website_form_recaptcha/data/ir_config_parameter_data.xml
Comment (lasley) [reply website_form_recaptcha/data/ir_config_parameter_data.xml]: Pretty sure this was my bad. Old IDE really loved tabs

File: website_form_recaptcha/README.rst
Comment (simahawk) [inline website_form_recaptcha/README.rst]: /11.0

File: website_form_recaptcha/__init__.py
Comment (simahawk) [inline website_form_recaptcha/__init__.py]: remove coding from py files

File: website_form_recaptcha/__manifest__.py
Comment (simahawk) [inline website_form_recaptcha/__manifest__.py]: https://github.com/OCA/website

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_form_recaptcha/README.rst
Comment (simahawk) [inline website_form_recaptcha/README.rst]: would be nice to mention which "website forms" are we talking about: "extends `website_form` core module" or something like that?

File: website_form_recaptcha/models/website_form_recaptcha.py
Comment (bouvyd) [reply website_form_recaptcha/models/website_form_recaptcha.py]: You're absolutely right :)

**Comment by bouvyd** [comment]
Review changes done :)

File: website_form_recaptcha/models/website_form_recaptcha.py
Comment (yajo) [inline website_form_recaptcha/models/website_form_recaptcha.py]: raising inside a loop? Weird...

From a UX perspective, changing to something like this would mean a big boost:

```py
msg = "\n".join(self._get_error_message(error) 
                for error in res.get('error-codes', []))
if msg:
    raise ValidationError(msg)
```

Comment (bouvyd) [reply website_form_recaptcha/models/website_form_recaptcha.py]: For me that's the limit between refactoring and migration; you're asking me to change the implementation...

I agree with you, of course; I'd rather have all error messages displayed at once.

Can be done ithen.

Comment (yajo) [reply website_form_recaptcha/models/website_form_recaptcha.py]: I guess migrations sometimes allow us to see problems that passed unnoticed before :blush:

Comment (hmrodrigues) [reply website_form_recaptcha/models/website_form_recaptcha.py]: hey @dbo-odoo any news?

Comment (hmrodrigues) [reply website_form_recaptcha/models/website_form_recaptcha.py]: @Yajo I don't think that @dbo-odoo will refactor the requested changes.
Can you accept the PR and I will do a new one with the refactoring?

Comment (yajo) [reply website_form_recaptcha/models/website_form_recaptcha.py]: Of course, I'm not blocking this PR for this reason, I'm blocking it for the :x: in Travis. We can't merge like that, sorry. Other checks may be red, but Travis and Runbot must be green or it would break the main branch. :confused:

Comment (pedrobaeza) [reply website_form_recaptcha/models/website_form_recaptcha.py]: Red status is due to this PR? Maybe rebasing changes the thing.

Comment (yajo) [reply website_form_recaptcha/models/website_form_recaptcha.py]: It's this PR, yes.

Comment (pedrobaeza) [reply website_form_recaptcha/models/website_form_recaptcha.py]: OK then, anyone can fix it?

Comment (hmrodrigues) [reply website_form_recaptcha/models/website_form_recaptcha.py]: @Yajo You are right. I didn't notice the red flag on Travis

Comment (hmrodrigues) [reply website_form_recaptcha/models/website_form_recaptcha.py]: @pedrobaeza I can do the job, but @dbo-odoo will need to accept my PR into is fork in order to update this PR or I'm wrong?

Comment (pedrobaeza) [reply website_form_recaptcha/models/website_form_recaptcha.py]: In this case, you can add a new PR directly to OCA that superseeds this, and we close this one.

Comment (hmrodrigues) [reply website_form_recaptcha/models/website_form_recaptcha.py]: @pedrobaeza new PR https://github.com/OCA/website/pull/458

**Comment by pedrobaeza** [comment]
Superseeded by #458

**Pull request closed without merging**
